### PR TITLE
Removed clang warnings

### PR DESCRIPTION
**Pull request by ahcorde**
Removed clang warnings related to this job https://ci.ros2.org/view/nightly/job/nightly_linux_clang_libcxx/1986/gcc/folder.-1649941887/

Introduced in this PR https://github.com/ros2/rclcpp/pull/2598

### CODE REVIEW

File: rclcpp/test/rclcpp/executors/test_reinitialized_timers.cpp [hunk 1]
```diff
@@ -32,7 +32,7 @@ class TestTimersLifecycle : public testing::Test
   void TearDown() override {rclcpp::shutdown();}
 };
 
-TYPED_TEST_SUITE(TestTimersLifecycle, ExecutorTypes);
+TYPED_TEST_SUITE(TestTimersLifecycle, ExecutorTypes, );
```
Comment (alsora) [inline rclcpp/test/rclcpp/executors/test_reinitialized_timers.cpp:35]: why this change? is it a typo?

Comment (ahcorde) [reply rclcpp/test/rclcpp/executors/test_reinitialized_timers.cpp:35]: no, it's another clang warning

Comment (christophebedard) [reply rclcpp/test/rclcpp/executors/test_reinitialized_timers.cpp:35]: It's specifically for: https://ci.ros2.org/view/nightly/job/nightly_linux_clang_libcxx/1986/gcc/folder.-1649941887/fileName.-411323968/

> must specify at least one argument for '...' parameter of variadic macro [-Wgnu-zero-variadic-macro-arguments] 35 | TYPED_TEST_SUITE(TestTimersLifecycle, ExecutorTypes); | ^

adding a comma (and nothing afterwards) does not change the behaviour and satisfies clang.

Comment (alsora) [reply rclcpp/test/rclcpp/executors/test_reinitialized_timers.cpp:35]: I agree with @fujitatomoya , the correct approach is to pass `ExecutorTypeNames` there.

**Comment by alsora** [comment]
@ahcorde it looks like the Rpr job is failing in the test that you modified.
It looks like the test gets run only for 2 of the 4 executors, and then it fails with:
```
-- run_test.py: return code -11
```

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21687/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15962/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1188/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22392/)

**Review by alsora — APPROVED** [review]
